docs/command_line_reference.rst

```diff
@@ -682,6 +683,60 @@ Client certificates can be presented regardless of the ``verify_certs`` setting,
 * Enable SSL, verify server certificates using private CA: ``--client-options="use_ssl:true,verify_certs:true,ca_certs:'/path/to/cacert.pem'"``
 * Enable SSL, verify server certificates using private CA, present client certificates: ``--client-options="use_ssl:true,verify_certs:true,ca_certs:'/path/to/cacert.pem',client_cert:'/path/to/client_cert.pem',client_key:'/path/to/client_key.pem'"``
 
+**Static Responses**
+
+Define a JSON file containing a list of objects with the following properties:
+
+* ``path``: A path or path pattern that should be matched. Only leading and trailing wildcards (``*``) are supported. A path containing only a wildcard acts matches any path.
+* ``body``: The respective response body.
+* ``body-encoding``: Either ``raw`` or ``json``. Use ``json`` by default and ``raw`` for the operation-type ``bulk`` and ``search``.
+
+Here we define the necessary responses for a track that bulk-indexes data::
+
+    [
+      {
+        "path": "*/_bulk",
+        "body": {
+          "errors": false,
+          "took": 1
+        },
+        "body-encoding": "raw"
+      },
+      {
+        "path": "/_cluster/health*",
+        "body": {
+          "status": "green",
+          "relocating_shards": 0
+        },
+        "body-encoding": "json"
+      },
+      {
+        "path": "/_all/_stats/_all",
+        "body": {
+          "_all": {
+            "total": {
+              "merges": {
+                "current": 0
+              }
+            }
+          }
+        },
+        "body-encoding": "json"
+      },
+      {
+        "path": "*",
+        "body": {},
+        "body-encoding": "json"
+      }
+    ]
+
+.. note::
+   Paths are evaluated from top to bottom. Therefore, place more restrictive paths at the top of the file.
+
+Save the above responses as ``responses.json`` and execute a benchmark as follows::
+
+    esrally race --track=geonames --challenge=append-no-conflicts-index-only --pipeline=benchmark-only --distribution-version=8.0.0 --client-options="static_responses:'responses.json'"
```

Comment:
It took me a few seconds to realize that whether that while specifying a `pipeline` it won't really target any ES. One might even use the default pipeline, ES will get launched but won't be used.

Should we clarify with a note here or above in the `client-options` section that when `static_responses:'file'` is used, the use should also specify the `benchmark-only` pipeline and the `distribution-version` as Rally doesn't have a way to derive the version automatically as it'd normally do.

Reply:
Thanks! I've added a note in https://github.com/elastic/rally/pull/1234/commits/37c0900542b1a088abf1c05869fad4462c222c84.